lib/thrift/generator/binary/framed/client.ex

```diff
@@ -41,6 +41,9 @@ defmodule Thrift.Generator.Binary.Framed.Client do
     bang_name = :"#{underscored_name}!"
     options_bang_name = :"#{underscored_options_name}!"
 
+    underscored_name = Macro.pipe(underscored_name, quote do unquote end, 0)
```

Comment:
@fishcakez, do you have any other suggestions on how to write this that involve a bit less macro magic?

Reply:
The standard way to solve this would be to use `bind_quoted`. However that requires binding variables, which makes the code in the generated modules much less clear:
```elixir
underscored_name = :the_fun_name
def unquote(underscored_name)(...) do
```
The more bound variables, the less clear the generated code becomes when using `Macro.to_string`.
 
Therefore we want to get around binding variables in the generated code. This means we need to include an `unquote/1` call in the AST which has a literal atom argument, and not a variable. If we use `unquote(var)` or `bind_quoted` we won't get a literal atom but a variable binding followed by an unquote call with variable argument. As `quote/1` is part of `Kernel.SpecialForms` we are limited in how we can do this. The easiest way I can think of to avoid creating the AST by hand is to use `quote/0` (which is not a Kernel.SpecialForm) and pipe into it with this function call.

Reply:
I had come up with a few approaches for this, but after consulting with @fishcakez he was able to find `Macro.pipe` to solve it, and we agreed it seemed the cleanest.
I'd say the next cleanest approach is:
```elixir
underscored_name = Code.string_to_quoted!(" unquote(:#{underscored_name})() ")
```
The string shows exactly the code that ends up getting generated, namely:
```elixir
defmacro unquote(:and)() do
  ...
end
```
But converting from a string to a quoted expression is not as good as just building the quoted expression directly with `Macro.pipe`, though I think both approaches are good ones.

@jparise do you have thoughts on what may work better here?

Reply:
I think what you have is fine. I was just fishing (ha!) for something more obvious.

Reply:
An alternative is to delay some of the metaprogramming to the generated file. So that we generate a file that calls macros. For example the file might look like:
```elixir
import Thift.Binary.Framed.Client.DSL

thrift_rpc :literal_underscored_name, ...
```
Of course we would lose the significant advantage of being able to see the explicit calls.